tests/unit/Request.ts

```diff
@@ -0,0 +1,391 @@
+		},
+
+		'filter builder': function () {
+			const filter = new store.Filter();
+			const betweenTwoAndFour = filter.gt('id', 2).lt('price', 5);
+			return runCollectionTest(store.filter(betweenTwoAndFour), {
+				queryParams: {
+					id: 'gt=2',
+					price: 'lt=5'
+				}
+			});
+		},
+
+		'filter builder ne or': function () {
+			const filter = new store.Filter();
+			const betweenTwoAndFour = filter.ne('id', 2).or(filter.eq('foo', true), filter.eq('foo'));
+			return runCollectionTest(store.filter(betweenTwoAndFour), {
+				queryParams: {
+					id: 'ne=2|foo=true|foo=undefined'
+				}
+			});
+		},
+
+		'filter relational': function () {
+			const filter = new store.Filter();
+			const innerFilter = new store.Filter().eq('foo', true);
+			const nestedFilter = filter[ 'in' ]('id', store.filter(innerFilter).select('id'));
+			return runCollectionTest(store.filter(nestedFilter), {
+				queryParams: {
+					id: 'in=(/mockRequest/?foo=true&select(id))'
+				}
+			});
+		},
+
+		'sort': function () {
+			const sortedCollection = store.sort({
+				property: 'prop1',
+				descending: true
+			}, {
+				property: 'prop2'
+			}, {
+				property: 'prop3',
+				descending: true
+			});
+			return runCollectionTest(sortedCollection, {
+				queryParams: {
+					'sort(-prop1,+prop2,-prop3)': ''
+				}
+			});
+		},
+
+		'sort with this.sortParam': function () {
+			store.sortParam = 'sort-param';
+
+			const sortedCollection = store.sort({
+				property: 'prop1',
+				descending: true
+			}, {
+				property: 'prop2'
+			}, {
+				property: 'prop3',
+				descending: true
+			});
+			return runCollectionTest(sortedCollection, {
+				queryParams: {
+					'sort-param': '-prop1,+prop2,-prop3'
+				}
+			});
+		},
+
+		'sort with different prefixes': function () {
+			store.descendingPrefix = '--';
+			store.ascendingPrefix = '++';
+
+			const sortedCollection = store.sort({
+				property: 'prop1',
+				descending: true
+			}, {
+				property: 'prop2'
+			}, {
+				property: 'prop3',
+				descending: true
+			});
+			return runCollectionTest(sortedCollection, {
+				queryParams: {
+					'sort(--prop1,++prop2,--prop3)': ''
+				}
+			});
+		},
+		'select': function () {
+			const selectCollection = store.select([ 'prop1', 'prop2' ]);
+			return runCollectionTest(selectCollection, {
+				queryParams: {
+					'select(prop1,prop2)': ''
+				}
+			});
+		},
+		'select with selectParam': function () {
+			store.selectParam = 'select-param';
+			const selectCollection = store.select([ 'prop1', 'prop2' ]);
+			return runCollectionTest(selectCollection, {
+				queryParams: {
+					'select-param': 'prop1,prop2'
+				}
+			});
+		},
+		'range': function () {
+			return runCollectionTest(store, { start: 15, end: 25 }, <any> {
+				queryParams: {
+					'limit(10,15)': ''
+				}
+			});
+		},
+		'range with rangeParam': function () {
+			store.rangeStartParam = 'start';
+			store.rangeCountParam = 'count';
+			return runCollectionTest(store, { start: 15, end: 25 }, <any> {
```

Comment:
These started needing an any cast when I upgraded to TS 1.6, and it was because they had a `queryParam` field, which doesn't exist in the `RequestOptions` type. Once I realized that, and changed them to use `query` it stopped needing the cast. Unfortunately this also revealed a bug in `UrlSearchParams` that is at least partially responsible for a couple new test case failures.